src/Repository/ExtractionResultRepository.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Repository;
+
+use App\Document\ExtractionResult;
+use Doctrine\ODM\MongoDB\DocumentManager;
+use Doctrine\ODM\MongoDB\Repository\DocumentRepository;
+
+/**
+ * Class ExtractionResultRepository.
+ */
+class ExtractionResultRepository extends DocumentRepository
+{
+    /**
+     * ExtractionResultRepository constructor.
+     *
+     * @param \Doctrine\ODM\MongoDB\DocumentManager $documentManager
+     *   The doctrine document manager
+     */
+    public function __construct(DocumentManager $documentManager)
+    {
+        $uow = $documentManager->getUnitOfWork();
+        $classMetaData = $documentManager->getClassMetadata(ExtractionResult::class);
```

Comment:
This magic might require more comments.

Reply:
Fixed

Reply:
It's still black magic. But what do it do? UnitOfWork related to entity manager ?

Reply:
Have added bigger comment